src/main/java/com/google/devtools/build/lib/rules/java/JavaCompileAction.java

```diff
@@ -489,6 +489,11 @@ public ActionResult execute(ActionExecutionContext actionExecutionContext)
           .insertDependencies(
               outputDepsProto,
               readFullOutputDeps(fallbackResults, actionExecutionContext, spawn.getPathMapper()));
+    } else if (!spawn.getPathMapper().isNoop()){
+      // As a side effect, readFullOutputDeps rewrites the on-disk .jdeps file from mapped to
+      // unmapped paths. To make path mapping fully transparent to consumers of this action's
+      // output, we ensure that the file always contains unmapped paths.
+      readFullOutputDeps(fallbackResults, actionExecutionContext, spawn.getPathMapper());
```

Comment:
I can't remember, but are there scenarios where Bazel would eagerly call this even if it's not needed?

i.e. is there risk of performance / memory degradation when path mapping is enabled because of this extra logic? And it applies to what may be numerous action results?

Reply:
I think that this call is always needed if path mapping is enabled, which is checked in the if condition: It's the only way to guarantee that the `.jdeps` output is correct. Since I would expect most compile actions to not fall back to the full classpath and those that don't already perform this action unconditionally in https://github.com/bazelbuild/bazel/pull/18155/files#diff-738efdd95b5eedf99d16e391201cb00e284b0ccf3542506f6c88cb01ffb96830R434, I wouldn't expect any regressions.